File: src/main/java/de/mikeyllp/miniGamesV4/commands/admincommands/setsubcommands/SetEnableDisableGame.java
package de.mikeyllp.miniGamesV4.commands.admincommands.setsubcommands;

import dev.jorel.commandapi.CommandAPICommand;
import dev.jorel.commandapi.arguments.ArgumentSuggestions;
import dev.jorel.commandapi.arguments.StringArgument;
import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.configuration.file.YamlConfiguration;
import org.bukkit.plugin.java.JavaPlugin;

import java.io.File;

import static de.mikeyllp.miniGamesV4.utils.MessageUtils.*;

public class SetEnableDisableGame extends CommandAPICommand {
    public SetEnableDisableGame(String commandName, JavaPlugin plugin) {

        super(commandName);


        withArguments(new StringArgument("miniGames").replaceSuggestions(ArgumentSuggestions.strings(
                "HideAndSeek", "TicTacToe", "RockPaperScissors", "SmallModus")));


        executes(((sender, arg) -> {
            String miniGameArg = (String) arg.get("miniGames");
            String msg = miniGameArg.trim().toLowerCase();

            // Get the Config
            FileConfiguration config = plugin.getConfig();

            switch (msg) {
                case "hideandseek":
                    if (config.getBoolean("HideAndSeek")) {
                        config.set("HideAndSeek", false);

                        // Save the config
                        plugin.saveConfig();

                        sendGameSwitch(sender, "HideAndSeek", false);
                        sendNeedReloadMessage(sender);
                    } else {
                        config.set("HideAndSeek", true);

                        // Save the config
                        plugin.saveConfig();

                        sendGameSwitch(sender, "HideAndSeek", true);
                        sendNeedReloadMessage(sender);
                    }
                    break;
                case "rockpaperscissors":
                    if (config.getBoolean("RockPaperScissors")) {
                        config.set("RockPaperScissors", false);

                        // Save the config
                        plugin.saveConfig();

                        sendGameSwitch(sender, "RockPaperScissors", false);
                        sendNeedReloadMessage(sender);
                    } else {
                        config.set("RockPaperScissors", true);

                        // Save the config
                        plugin.saveConfig();

                        sendGameSwitch(sender, "RockPaperScissors", true);
                        sendNeedReloadMessage(sender);
                    }
                    break;
                case "tictactoe":
                    if (config.getBoolean("TicTacToe")) {
                        config.set("TicTacToe", false);

                        // Save the config
                        plugin.saveConfig();

                        sendGameSwitch(sender, "TicTacToe", false);
                        sendNeedReloadMessage(sender);
                    } else {
                        config.set("TicTacToe", true);

                        // Save the config
                        plugin.saveConfig();

                        sendGameSwitch(sender, "TicTacToe", true);
                        sendNeedReloadMessage(sender);
                    }
                    break;
                case "smallmodus":
                    if (config.getBoolean("small-modus")) {
                        config.set("small-modus", false);

                        // Save the config
                        plugin.saveConfig();

                        sendGameSwitch(sender, "small-modus", false);
                        sendNeedReloadMessage(sender);
                    } else {
                        config.set("small-modus", true);

                        // Save the config
                        plugin.saveConfig();

                        sendGameSwitch(sender, "small-modus", true);
                        sendNeedReloadMessage(sender);
                    }
                    break;
                default:
                    String lang = plugin.getConfig().getString("lang");
                    File file = new File(plugin.getDataFolder(), "languages/" + lang + ".yml");
                    sendCustomMessage(sender, YamlConfiguration.loadConfiguration(file).getString("warning-message.false-game"));
            }
        }));
    }
}

File: src/main/java/de/mikeyllp/miniGamesV4/commands/admincommands/setsubcommands/SetEnableDisableGame.kt
package de.mikeyllp.miniGamesV4.commands.admincommands.setsubcommands

import de.mikeyllp.miniGamesV4.messages.MessageUtils
import de.mikeyllp.miniGamesV4.messages.Translator
import de.mikeyllp.miniGamesV4.plugin
import dev.jorel.commandapi.CommandAPICommand
import dev.jorel.commandapi.arguments.ArgumentSuggestions
import dev.jorel.commandapi.arguments.StringArgument
import dev.jorel.commandapi.kotlindsl.anyExecutor
import dev.jorel.commandapi.kotlindsl.subcommand
import java.util.*

fun CommandAPICommand.setEnableDisableGame() = subcommand("invert") {
    withArguments(
        StringArgument("miniGames").replaceSuggestions(
            ArgumentSuggestions.strings(
                "HideAndSeek", "TicTacToe", "RockPaperScissors", "SmallModus"
            )
        )
    )

    anyExecutor { sender, args ->
        val miniGameArg = args["miniGames"] as String
        val msg: String = miniGameArg.trim { it <= ' ' }.lowercase(Locale.getDefault())
        val config = plugin.config

        when (msg) {
            "hideandseek" -> if (config.getBoolean("HideAndSeek")) {
                config.set("HideAndSeek", false)

                plugin.saveConfig()

                MessageUtils.sendGameSwitch(sender, "HideAndSeek", false)
                MessageUtils.sendNeedReloadMessage(sender)
            } else {
                config.set("HideAndSeek", true)

                plugin.saveConfig()

                MessageUtils.sendGameSwitch(sender, "HideAndSeek", true)
                MessageUtils.sendNeedReloadMessage(sender)
            }

            "rockpaperscissors" -> if (config.getBoolean("RockPaperScissors")) {
                config.set("RockPaperScissors", false)

                plugin.saveConfig()

                MessageUtils.sendGameSwitch(sender, "RockPaperScissors", false)
                MessageUtils.sendNeedReloadMessage(sender)
            } else {
                config.set("RockPaperScissors", true)

                plugin.saveConfig()

                MessageUtils.sendGameSwitch(sender, "RockPaperScissors", true)
                MessageUtils.sendNeedReloadMessage(sender)
            }

            "tictactoe" -> if (config.getBoolean("TicTacToe")) {
                config.set("TicTacToe", false)

                plugin.saveConfig()

                MessageUtils.sendGameSwitch(sender, "TicTacToe", false)
                MessageUtils.sendNeedReloadMessage(sender)
            } else {
                config.set("TicTacToe", true)

                plugin.saveConfig()

                MessageUtils.sendGameSwitch(sender, "TicTacToe", true)
                MessageUtils.sendNeedReloadMessage(sender)
            }

            "smallmodus" -> if (config.getBoolean("small-modus")) {
                config.set("small-modus", false)

                plugin.saveConfig()

                MessageUtils.sendGameSwitch(sender, "small-modus", false)
                MessageUtils.sendNeedReloadMessage(sender)
            } else {
                config.set("small-modus", true)

                plugin.saveConfig()

                MessageUtils.sendGameSwitch(sender, "small-modus", true)
                MessageUtils.sendNeedReloadMessage(sender)
            }

            else -> {
                MessageUtils.sendMessage(sender, Translator.translatable("warning-message.false-game"))
            }
        }
    }
}
